templates/password_pages/password_required.html

```diff
@@ -0,0 +1,52 @@
+{% extends 'base_page.html' %}
+
+{% load static wagtailmetadata_tags wagtailimages_tags wagtailcore_tags i18n %}
+{% block meta_tag %}
+  {% meta_tags %}
+{% endblock %}
+
+{% if breadcrumb %}
+    {% include 'includes/breadcrumb.html' %}
+{% endif %}
+
+{% block content %}
+    <div class="container">
+        <div class="row">
+            <div class="col-12">
+                <h1>Password required</h1>
+                <p>
+                    You need a password to access this page.
+                </p>
+                <form action="{{ action_url }}" method="POST" class="tna-form password-form">
+                    {% csrf_token %}
+
+                    {{ form.non_field_errors }}
+
+                    <div class="password-form__password">
+                        {{ form.password.errors }}
+                        <label class="password-form__label" for="id_password">Password</label>
+                        <input class="password-form__text" type="password" name="password" required id="id_password" placeholder="Your password">
+                        <input class="password-form__show" type="checkbox" onclick="myFunction()">
+                    </div>
+
+                    {% for field in form.hidden_fields %}
+                        {{ field }}
+                    {% endfor %}
+                    <input class="tna-button--dark password-form__submit" type="submit" value="View page" />
+                </form>
+            </div>
+        </div>
+    </div>
+
+    <script type="text/javascript">
+        function myFunction() {
+            const x = document.getElementById("id_password");
+            if (x.type === "password") {
+                x.type = "text";
+            } else {
+                x.type = "password";
+            }
+        }
```

Comment:
```suggestion
        function switchPasswordVisibility(event) {
            var x = document.getElementById("id_password");
            if (event.currentTarget.checked) {
                x.type = "password";
            } else {
                x.type = "text";
            }
        }
        
        var showHidePasswordCheckbox = document.createElement("input");
        showHidePasswordCheckbox.classList.add("password-form__show");
        showHidePasswordCheckbox.setAttribute("type", "checkbox");
        showHidePasswordCheckbox.addEventListener("change", switchPasswordVisibility);
        
        var formPasswordContainer = document.querySelector(".password-form__password");
        formPasswordContainer.appendChild(showHidePasswordCheckbox);
```

Reply:
This will (I hope) only add the show/hide password checkbox when JS is available rather than having it always available and do nothing when JS doesn't work or is disabled.